[Max Autotune][B200] Relax absolute tolerance for MM+MM test

jananisriram opened the pull request:
Summary: Relax absolute tolerance from 1e-2 to 1e-1 for `test_non_contiguous_input_mm_plus_mm` in `test_max_autotune.py`.

Test Plan: `test_max_autotune.py`

Differential Revision: D83391942




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

eellison commented on torch/testing/_internal/common_utils.py:
```diff
@@ -5756,6 +5756,31 @@ def recover():
 
     return recover()(fn)
 
+def patch_matmul(**new_flags):
```
I think this is equivalent to `unittest.mock.patch `

jananisriram replied:
thanks! when i run with `@mock.patch`, i actually get this error:
`AttributeError: 'cuBLASModule' object has no attribute 'allow_bf16_reduced_precision_reduction'`

this isn't critical for this pr to merge, but would make the test cleaner; can push this to a future pr!

jananisriram committed:
[Max Autotune][B200] Fix decompose_k test failure (#164021)

Summary:

Fix decompose_k test failure (`test_max_autotune_decompose_k `) in `test_max_autotune.py` on B200s by setting `torch._inductor.config` patches for variables `comprehensive_padding` and `shape_padding`. Initial failure was `AssertionError: False is not true : Could not find a split in {3, 9, 2187, 81, 243, 729, 27} in # AOT ID: ['6_forward']`. 

Refactor decompose_k test to follow patch semantics when setting all environment variables within a test.

Test Plan:
`test_max_autotune.py`:
```
buck2 test 'fbcode//mode/opt' fbcode//caffe2/test/inductor:max_autotune -c fbcode.nvcc_arch=b200a -c fbcode.enable_gpu_sections=true -c fbcode.platform010_cuda_version=12.8 -c fbcode.re_gpu_tests=False -- test_max_autotune_decompose_k
```

Reviewed By: eellison, njriasan

Differential Revision: D83390563

jananisriram committed:
[Max Autotune][B200] Relax tolerances for MM+MM and bAddMM tests (#164022)

Summary:

Relax absolute tolerance from 1e-2 to 1e-1 for `test_non_contiguous_input_mm_plus_mm` in `test_max_autotune.py`. Relax relative tolerance from 1e-3 to 2e-3 for `test_baddmm`.

Test Plan: `test_max_autotune.py`

Reviewed By: eellison

Differential Revision: D83391942
